### PR TITLE
INGK-464: CanopenDictionary class is updated

### PR DESCRIPTION
**Pull request by WoodElv**

### CODE REVIEW

File: tests/dictionary/test_canopen_dictionary.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import pytest
+from tests.conftest import read_config
+from ingenialink.canopen.dictionary import CanopenDictionary
+
+
+@pytest.mark.smoke
+def test_get_register_dictionary(read_config):
+    file_test_path = read_config['canopen']['dictionary']
```
Comment (GullinBustin) [inline tests/dictionary/test_canopen_dictionary.py:8]: That should be a test dictionary, if you make it configurable could fail, for example, if the number of registers is different.

**Review by GullinBustin — CHANGES_REQUESTED** [review]

File: ingenialink/canopen/dictionary.py [hunk 2]
```diff
@@ -157,97 +26,19 @@ def read_register(self, register):
...
-        Args:
-            subnode (int): Identifier for the subnode.
-
-        Returns:
-            dict: Dictionary of all the registers for a subnode.
-
-        """
-        return self.__registers[subnode]
+        except Exception:
+            pass
```
Comment (GullinBustin) [inline ingenialink/canopen/dictionary.py:44]: This exception is very dangerous, can hide a lot of bugs

**Review by GullinBustin — CHANGES_REQUESTED** [review]

File: tests/dictionary/test_canopen_dictionary.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import pytest
+from tests.conftest import read_config
```
Comment (MartinALL91) [inline tests/dictionary/test_canopen_dictionary.py:2]: Leave a blank space between each group of imports.

File: tests/dictionary/test_canopen_dictionary.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+from ingenialink.canopen.dictionary import CanopenDictionary
+
+
+@pytest.mark.smoke
+def test_get_register_dictionary(read_config):
+    file_test_path = read_config['canopen']['dictionary']
+    canopen_dict = CanopenDictionary(file_test_path)
+
+    assert canopen_dict.subnodes == 2
```
Comment (MartinALL91) [inline tests/dictionary/test_canopen_dictionary.py:11]: Use  SINGLE_AXIS_MINIMUM_SUBNODE

File: tests/dictionary/test_canopen_dictionary.py [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+    regs_sub0 = canopen_dict.registers(0)
+    assert len(regs_sub0) == 167 - 1
+    regs_sub1 = canopen_dict.registers(1)
+    assert len(regs_sub1) == 510
```
Comment (MartinALL91) [inline tests/dictionary/test_canopen_dictionary.py:16]: Make this test generic (It should work with any dictionary)

File: ingenialink/canopen/dictionary.py [hunk 6]
```diff
@@ -92,162 +20,52 @@ class CanopenDictionary(Dictionary):
...
-        """
-        return self.__registers[subnode]
+            current_read_register = super()._read_register(register)
+
+            current_read_register[AttrRegCanDict.IDX] = int(register.attrib['address'][:6], 16)
+            current_read_register[AttrRegCanDict.SUBIDX] = int("0x" + register.attrib['address'][-2:], 16)
+
+            return current_read_register
+
+        except KeyError as ke:
```
Comment (GullinBustin) [inline ingenialink/canopen/dictionary.py:44]: As in INGK-463: This try-except changes the function behavior. The parent function can't return None, but this function does.

Comment (WoodElv) [reply ingenialink/canopen/dictionary.py:44]: It's solved here: [INGK-464](https://github.com/ingeniamc/ingenialink-python/pull/213#discussion_r996689087)

**Review by GullinBustin — APPROVED** [review]

File: ingenialink/canopen/dictionary.py [hunk 7]
```diff
@@ -1,86 +1,9 @@
-import xml.etree.ElementTree as ET
-from ..dictionary import Dictionary
-from ..constants import SINGLE_AXIS_MINIMUM_SUBNODES
-from .register import CanopenRegister, REG_ACCESS, REG_DTYPE
-from ingenialink.utils._utils import *
-from .._ingenialink import lib
+from ingenialink.dictionary import Dictionary
+from ingenialink.constants import SINGLE_AXIS_MINIMUM_SUBNODES
+from ingenialink.canopen.register import CanopenRegister
+import ingenialogger
```
Comment (MartinALL91) [inline ingenialink/canopen/dictionary.py:4]: This is a third-party library. It should be on top of the local library imports and separated with a blank line.

**Review by MartinALL91 — APPROVED** [review]

**Pull request merged**